### DISCUSSION TITLE
QST: should sjoin lsuffix and rsuffix default same as pandas join?

### DISCUSSION THREAD

**Discussion by raybellwaves** [geopandas/geopandas#2708] [Q&A]
I really enjoy that the API is similar to pandas. It's really helped myself get spun up with this package.

I'm curious if the default args of lsuffix and rsuffix of [sjoin](https://geopandas.org/docs/reference/api/geopandas.sjoin.html) should match that of [pandas.DataFrame.join](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.join.html)? That is "" and "" instead of "left" and "right".

I understand this is a minor thing as they are changeable args but it would be nice to match defaults values along with the API (regardless if those default values aren't ideal in pandas)

**Comment by raybellwaves** [comment]
ah this is trickier that I thought

```
>>> geopandas.sjoin(cities, countries, how="inner", op='intersects', lsuffix="", rsuffix="")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ray/local/bin/anaconda3/envs/main/lib/python3.8/site-packages/geopandas/tools/sjoin.py", line 91, in sjoin
    joined = _frame_join(indices, left_df, right_df, how, lsuffix, rsuffix)
  File "/home/ray/local/bin/anaconda3/envs/main/lib/python3.8/site-packages/geopandas/tools/sjoin.py", line 258, in _frame_join
    left_df.merge(indices, left_index=True, right_index=True)
  File "/home/ray/local/bin/anaconda3/envs/main/lib/python3.8/site-packages/pandas/core/frame.py", line 4727, in set_index
    raise KeyError(f"None of {missing} are in the columns")
KeyError: "None of ['index_'] are in the columns"
```

**Comment by martinfleis** [comment]
The suffixes are used internally in the function at the moment so they cannot be the same but I agree that it may be better to have consistency with pandas here. However, it would require a bit of refactoring for a fairly small difference so it is a change of a low priority.

In other words - I'd be fine with the change (with some deprecation period) if someone wants to work on it.

**Comment by jschuba** [comment]
I thought I should point out another API difference with Geopandas vs Pandas suffixes.  Geopandas tacks on an extra "_", even when you explicitly ask for no suffix.   I use this technique in Pandas all the time, to make sure my left DataFrame columns are not altered, but I can't do it in Geopandas.  

Here is code to demonstrate: 

```
import pandas as pd
import geopandas as gpd

data = pd.DataFrame([
    {'id': 1,
     'longitude': 54.606402,
     'latitude': 18.347709},
    {'id': 2,
     'longitude': 54.604681,
     'latitude': 18.346534},
    {'id': 3,
     'longitude': 54.544322,
     'latitude': 18.293896},
    {'id': 4,
     'longitude': 50.843233,
     'latitude': 11.130688}
])

left = gpd.GeoDataFrame(data, geometry=gpd.points_from_xy(data.longitude, data.latitude), crs='epsg:4326')
right = gpd.GeoDataFrame(data, geometry=gpd.points_from_xy(data.longitude, data.latitude), crs='epsg:4326')

merged_gpd = gpd.sjoin(left, right, lsuffix='', rsuffix='_right')  # Notice Suffixes

merged_pandas = pd.merge(left, right, on=['latitude', 'longitude'], suffixes=('', '_right'))  # Notice Suffixes
```

We see that Geopandas turns the left-side "id" into "id_", whereas Pandas does not alter this column. 

```
print(list(merged_gpd.columns))
['id_', 'longitude_', 'latitude_', 'geometry', 'index__right', 'id__right', 'longitude__right', 'latitude__right']
print(list(merged_pandas.columns))
['id', 'longitude', 'latitude', 'geometry', 'id_right', 'geometry_right']
```

**Comment by martinfleis** [comment]
>  Geopandas tacks on an extra "_", even when you explicitly ask for no suffix.

I wouldn't call that an API difference, but a bug :). Would you be willing to try a pushing a fix?
